File: leetcode-submissions/1302_Deepest_Leaves_Sum.ts
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     val: number
 *     left: TreeNode | null
 *     right: TreeNode | null
 *     constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.left = (left===undefined ? null : left)
 *         this.right = (right===undefined ? null : right)
 *     }
 * }
 */

function deepestLeavesSum(root: TreeNode | null): number {
    if(!root) return 0;
    const q:Array<TreeNode|null>=[root];
    const ans:number[][]=[];
    while(q.length>0){
let size:number =q.length;
const temp:number[]=[];
while(size--){
    const currNode:TreeNode|null=q.shift()!;
    temp.push(currNode.val);
    if(currNode.left){
        q.push(currNode.left);
    }
    if(currNode.right){
        q.push(currNode.right);
    }
}
ans.push(temp);
    }
    let sum:number =0;
    for(let i=0;i<ans[ans.length-1].length;i++){
sum+=ans[ans.length-1][i];
    }
    return sum ;
};

File: leetcode-submissions/1302_Deepest_Leaves_Sum.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int deepestLeavesSum(TreeNode* root) {
        queue<TreeNode*>q;
        q.push(root);
        vector<vector<int>>ans ;
        while(not q.empty()){
            int currSize=q.size();
            vector<int>temp;

            while(currSize--){
                TreeNode* currNode =q.front();
                q.pop();
                temp.push_back(currNode->val);
                if(currNode->left){
                    q.push(currNode->left);

                }
                 if(currNode->right){
                    q.push(currNode->right);

                }

            }
            ans.push_back(temp);
        }
        int DeapestLeaveSum=0;
        vector<int>lastlevelNodes =ans[ans.size()-1];
        for(int i=0;i<lastlevelNodes.size();i++){
            DeapestLeaveSum+=lastlevelNodes[i];
        }
        return DeapestLeaveSum;
    }
};